### PR TITLE
Eslint

### PR DESCRIPTION
**Pull request by etrobert**
Add eslint to the project with default configuration

Fix the errors and warnings in the project

### CODE REVIEW

File: src/scripts/graph.ts [hunk 1]
```diff
@@ -209,26 +209,34 @@ function resetSelected() {
...
 
-export function getGraph() {
+export function getGraph(): Graph {
   const tasksHtml = getTasks();
   const tasks = tasksHtml.map((e) => {
     const bb = getOffsetBox(e);
     return {
-      name: e.textContent,
+      // We assume tasks have a textContent
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (justinline) [inline src/scripts/graph.ts:224]: question: could you also solve this by making `textContent` required on the `HTMLTaskElement` type? 🤔 hard to tell in this diff

File: src/scripts/graph.ts [hunk 2]
```diff
@@ -209,26 +209,34 @@ function resetSelected() {
...
+      status: e.classList.contains("completed")
+        ? ("completed" as const)
+        : ("todo" as const),
     };
   });
   const dependenciesHtml = getDependencies();
   const dependencies = dependenciesHtml.map((e) => ({
-    predecessor: e.from.textContent,
-    successor: e.to.textContent,
+    // We assume dependencies and tasks are correctly set
```
Comment (justinline) [inline src/scripts/graph.ts:234]: thought: lot's of assumptions 😛 😨

File: src/scripts/graph.ts [hunk 3]
```diff
@@ -251,7 +261,9 @@ function updatePath(path: HTMLDependencyElement, dest?: Point) {
   );
   const pathPointB = dest
     ? dest
-    : intersectLineBox(centerA, centerB, getExpandedBox(nodeBBox!, offset));
+    : // If dest is undefined then nodeBBox is not null
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (justinline) [inline src/scripts/graph.ts:265]: thought: seems weird that typescript cannot infer this

suggestion: it's also a little hard to follow the 3 ternaries all using `dest` but I know it's a bit out of scope for the MR

**Comment by justinline** [comment]
praise: well done typing all that 👍

**Comment by justinline** [comment]
Finished 🏁

File: src/scripts/graph.ts (hunk 1 above)
Comment (etrobert) [reply src/scripts/graph.ts:224]: That's a great idea, I will do that! 😌

That being said, this `HTMLTaskElement` with custom `from` and `to` properties is completely illegal and will be taken down ASAP 😁

File: src/scripts/graph.ts (hunk 2 above)
Comment (etrobert) [reply src/scripts/graph.ts:234]: Fixed by your `HTMLTaskElement` suggestion 😌

File: src/scripts/graph.ts (hunk 3 above)
Comment (etrobert) [reply src/scripts/graph.ts:265]: I know this piece of code is awful 🙈

I remember I didn't know how to organise it better at the time

I just tried: I still don't 😅 

The only valid idea coming to my mind would be to have two distinct function with huge amount of code duplication but that's no good either 😞

If a way seems obvious to you I'd be happy to hear it 🤗

**Comment by etrobert** [comment]
Thank you for your time ❤️
